debug_module.tex

```diff
@@ -167,6 +167,15 @@ \section{Abstract Commands} \label{abstractcommands}
 Debuggers can only determine which abstract commands
 are supported by a given hart in a given state by attempting them
 and then looking at \Fcmderr in \Rabstractcs to see if they were successful.
+Commands may be supported with some options set, but not with other options
+set. If a command has unsupported options set, the DM must set \Fcmderr to 2
+(not supported).
+
+\begin{commentary}
+    Example: A system with no Program Buffer must support the Access Register
+    command, but will still return "not supported" when the Access Register
+    command has \Fpostexec set.
```

Comment:
this is not an example i would have expected... if you don’t have program buffer I figured you just had no program buffer to write and maybe you would treat it like implicit ebreak. That’s actually a more interesting error case and we should actually define it? Could people execute all-zero instructions on the nonexistent program buffer if they have nonimplicit ebreak?

I would have thought a good example here is if eg CSR abstract access is not supported

Reply:
As far as I know, the postexec attribute has been removed from the Access Memory abstract command.  So, this example doesn't seem to apply any longer.

Reply:
I read it as the access register command (the commentary was a more general statement about all abstract commands, not specific to access memory)

Reply:
I'm changing this to the CSR example, and have filed https://github.com/riscv/riscv-debug-spec/issues/301 to discuss what happens with postexec when there's no program buffer.